src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -432,6 +553,48 @@ public final Builder sound(SoundEvent fillSound, SoundEvent emptySound)
             return this;
         }
 
+        public final Builder motionScale(ToDoubleBiFunction<FluidState, Entity> motionScale)
+        {
+            this.motionScale = motionScale;
+            return this;
+        }
+
+        public final Builder fallDistanceModifier(ToDoubleBiFunction<FluidState, Entity> fallDistanceModifier)
+        {
+            this.fallDistanceModifier = fallDistanceModifier;
+            return this;
+        }
+
+        public final Builder canSwim(Predicate<FluidState> canSwim)
+        {
+            this.canSwim = canSwim;
+            return this;
+        }
+
+        public final Builder canDrown(BiPredicate<FluidState, LivingEntity> canDrown)
+        {
+            this.canDrown = canDrown;
+            return this;
+        }
+
+        public final Builder canExtinguish(BiPredicate<FluidState, Entity> canExtinguish)
+        {
+            this.canExtinguish = canExtinguish;
+            return this;
+        }
+
+        public final Builder canHydrate(BiPredicate<FluidState, BlockState> canHydrate)
+        {
+            this.canHydrate = canHydrate;
+            return this;
+        }
+
+        public final Builder canBoat(BiPredicate<FluidState, Boat> canBoat)
+        {
+            this.canBoat = canBoat;
+            return this;
+        }
```

Comment:
Given there are already javadocs for what these params are for, it may be helpful to also add them on these builder methods as the brief description of what it is for.